library/sha512.c

```diff
@@ -51,6 +51,14 @@
 #endif /* MBEDTLS_PLATFORM_C */
 #endif /* MBEDTLS_SELF_TEST */
 
+#if defined(MBEDTLS_PLATFORM_C)
+#include "mbedtls/platform.h"
+#else
+#include <stdlib.h>
+#define mbedtls_calloc    calloc
+#define mbedtls_free       free
+#endif
```

Comment:
Once more, this block of code is very similar to lines 46-51. Perhaps is better to just add the missing proprocessor directives there. Also, the only function modified is a self_test, so there should not be any problems.